plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererFeatureConfigService.kt

```diff
@@ -84,20 +92,24 @@ class CodeWhispererFeatureConfigService {
 
     fun getCustomizationArnOverride(): String = getFeatureValueForKey(CUSTOMIZATION_ARN_OVERRIDE_NAME).stringValue()
 
+    fun getNewAutoTriggerUX(): Boolean = getFeatureValueForKey(NEW_AUTO_TRIGGER_UX).boolValue()
+
     // Get the feature value for the given key.
-    // In case of a misconfiguration, it will return a default feature value of Boolean true.
+    // In case of a misconfiguration, it will return a default feature value of Boolean false.
     private fun getFeatureValueForKey(name: String): FeatureValue =
         featureConfigs[name]?.value ?: FEATURE_DEFINITIONS[name]?.value
-            ?: FeatureValue.builder().boolValue(true).build()
+            ?: FeatureValue.builder().boolValue(false).build()
 
     companion object {
         fun getInstance(): CodeWhispererFeatureConfigService = service()
         private const val TEST_FEATURE_NAME = "testFeature"
         private const val DATA_COLLECTION_FEATURE = "IDEProjectContextDataCollection"
         const val CUSTOMIZATION_ARN_OVERRIDE_NAME = "customizationArnOverride"
+        private const val NEW_AUTO_TRIGGER_UX = "newAutoTriggerUX"
```

Comment:
nit: we could make these constants sealed data object

Reply:
like
```
sealed interface Feature {
     val  id: String
     
     sealed data object Feature1 { id: 'feat1'}

     sealed data object Feature2 { id: 'feat2' }
}
```

Reply:
can do this later

Reply:
ok